docs/proposals/2018-06-04_Collections_Promo_References_Aliases.md

```diff
@@ -0,0 +1,156 @@
+# Collections initial feature set and schema
+
+## Problem
+
+To date, ANS has no dedicated way of representing curated lists of content to be displayed in sequence.  Some users have used the `content_elements` field with on a story document to receive full inflation of sub-documents, and others have used `related_content` in a similar way.  However, both are ad hoc implementations to get around the lack of the feature rather than an intended use case of an ANS feature directly.  As a result, these implementations have been subject to restrictions and limitations that make them less than ideal.
+
+This document suggests several new ANS features to address this problem directly.
+
+## New ANS Feature: Promo References
+
+A new object type will be added to ANS with an accompanying feature in Content API: the promo reference.
+
+A promo reference will have the same form as any other reference, but with a few key differences:
+ * Unlike regular references, promo references will resolve against a *denormalized* document view, and therefore may include additional denormalized content.
+
+ * Promo references will not count against the normal reference limit. (Content API currently rejects documents with more than 300 discrete references.) A single document will be limited to 500 promo references.
+
+ * Promo references will only be denormalized at query time. They will not be denormalized when written onto the Content Event Stream.
+
+ * A promo reference is denormalized into an object with only the following fields:
+   * _id
+   * version
+   * type
+   * subtype
+   * headlines
+   * subheadlines
+   * description
+   * label
+   * promo_items
+   * credits
+   * taxonomy
+   * websites
+   * canonical_website
+   * canonical_url
+   * short_url
+   * display_date
+   * publish_date
+   * language
+   * alignment
+   * (...additional fields may be added at a later time.)
+
+ * Promo references only reference story, gallery and video documents.  Additional reference types may be added at a later time.
+
+ * Promo references have the same form as a regular reference in most other respects, but their type is `promo_reference`.
+
+
+### New ANS Content Type: Collections
+
+A new document type `collection` represents in ANS a curated sequence of content. A collection has the same ANS fields as a `story`. However, collections have some significant behavioral differences.
+
+    * `content_elements` in a `collection` may only consist of promo references
```

Comment:
Is `collection` then not an overly generic term for this? Can we not think of any potential future scenarios where we'd want to represent an ordered collection of items/references other than this specific use case?  I wonder if `promo_collection` (or something along those lines) wouldn't be a better name.

Reply:
I think the product name "Collections" is set in stone at this point, but I'll verify

Reply:
So it's not set in stone -- but it's the most likely name at this point. 

As for being overly generic, collections are indeed intended to cover a wide variety of use cases, so the name is necessarily generic.  (Lists on homepages, "story galleries", promo boxes, etc)

If we need a different type down the line that is more specific, or that is also somewhat generic, I think we still have "lists", "sets", "sequences", "featured_items_list" etc available to us at that time.

Reply:
Ok. Personally I still find it overly generic, because I can think of lots of use cases for an ordered collection of items which is not restricted to `promo_item`, and I don’t see the harm in specificity—but I also don’t feel that strongly about it and mostly just wanted to raise and make sure we considered. So if you’re good with it then I’m good with it.